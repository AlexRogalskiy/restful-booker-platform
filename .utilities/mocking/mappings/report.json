{
  "request": {
    "method": "GET",
    "url": "/report/"
  },
  "response": {
    "status": 200,
    "headers" : {
      "Vary": "Origin",
      "Vary": "Access-Control-Request-Method",
      "Vary": "Access-Control-Request-Headers",
      "Access-Control-Allow-Origin": "http://localhost:3003",
      "Access-Control-Allow-Credentials": "true",
      "Content-Type": "application/json;charset=UTF-8",
      "Transfer-Encoding": "chunked"

    },
    "body": "{\"rooms\":[101], \"totals\":[100]}"
  }
}
