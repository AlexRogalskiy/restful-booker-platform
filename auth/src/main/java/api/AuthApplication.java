package api;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.ComponentScan;

@SpringBootApplication
@ComponentScan()
public class AuthApplication {

    // Testing the lookup by making a change in search
    public static void main(String[] args) {
        SpringApplication.run(AuthApplication.class, args);
    }

}
