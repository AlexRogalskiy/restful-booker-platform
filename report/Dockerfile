FROM maven:3.5.2-jdk-8-alpine

ADD . /usr/local/report

WORKDIR /usr/local/report

COPY target ./

ENV mysqlDomain=mysql

ENTRYPOINT java -jar target/restful-booker-platform-report-1.0-SNAPSHOT.jar -D