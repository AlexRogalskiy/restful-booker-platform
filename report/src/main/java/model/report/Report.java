package model.report;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class Report {

    @JsonProperty
    private List<RoomReport> report;

    public Report(List<RoomReport> report) {
        this.report = report;
    }

    public List<RoomReport> getReport() {
        return report;
    }

    public void setReport(List<RoomReport> report) {
        this.report = report;
    }
}
