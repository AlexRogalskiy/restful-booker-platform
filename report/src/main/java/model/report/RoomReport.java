package model.report;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class RoomReport {

    @JsonProperty
    private String room;

    @JsonProperty
    private List<RoomReportDate> values;

    public RoomReport(String room, List<RoomReportDate> values) {
        this.room = room;
        this.values = values;
    }

    public String getRoom() {
        return room;
    }

    public void setRoom(String room) {
        this.room = room;
    }

    public List<RoomReportDate> getValues() {
        return values;
    }

    public void setValues(List<RoomReportDate> values) {
        this.values = values;
    }
}
