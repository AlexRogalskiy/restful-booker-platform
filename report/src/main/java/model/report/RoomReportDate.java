package model.report;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Date;

public class RoomReportDate {

    @JsonProperty()
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private Date date;

    public RoomReportDate(Date date) {
        this.date = date;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    @Override
    public String toString() {
        return "RoomReportDate{" +
                "date=" + date +
                '}';
    }
}
