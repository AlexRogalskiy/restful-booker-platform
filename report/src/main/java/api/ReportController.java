package api;

import model.report.Entry;
import model.report.Report;
import model.room.Booking;
import model.room.Bookings;
import model.room.Room;
import org.springframework.context.annotation.Bean;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
import requests.BookingRequests;
import requests.RoomRequests;


import java.sql.SQLException;
import java.util.*;

@RestController
public class ReportController {

    private RoomRequests roomRequests;
    private BookingRequests bookingRequests;

    @Bean
    public WebMvcConfigurer corsConfigurer() {
        return new WebMvcConfigurer() {
            @Override
            public void addCorsMappings(CorsRegistry registry) {
                String originHost = "http://localhost:3003";

                if(System.getenv("cors") != null){
                    originHost = System.getenv("cors");
                }

                registry.addMapping("/*")
                        .allowedOrigins(originHost)
                        .allowCredentials(true);
            }
        };
    }

    public ReportController() throws SQLException {
        roomRequests = new RoomRequests();
        bookingRequests = new BookingRequests();
    }

    @RequestMapping(value = "/", method = RequestMethod.GET)
    public Report getReport() {
        List<Room> rooms = roomRequests.searchForRooms().getBody().getRooms();
        List<Entry> parsedRooms = new ArrayList<>();

        for(Room r : rooms){
            Bookings roomBookings = bookingRequests.getBookings(r.getRoomid()).getBody();

            for(Booking b : roomBookings.getBookings()){
                Entry entry = new Entry(b.getBookingDates().getCheckin(), b.getBookingDates().getCheckout(), b.getFirstname() + " " + b.getLastname() + " - Room: " + r.getRoomNumber());
                parsedRooms.add(entry);
            }
        }

        return new Report(parsedRooms);
    }

}
