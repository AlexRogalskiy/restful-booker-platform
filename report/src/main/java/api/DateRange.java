package api;

import model.report.RoomReportDate;

import java.util.Calendar;
import java.util.Date;
import java.util.GregorianCalendar;
import java.util.List;

public class DateRange {

    public static List<RoomReportDate> parse(List<RoomReportDate> dates, Date startdate, Date enddate) {
        Calendar calendar = new GregorianCalendar();
        calendar.setTime(startdate);

        while (calendar.getTime().before(enddate)) {
            Date result = calendar.getTime();
            dates.add(new RoomReportDate(result));
            calendar.add(Calendar.DATE, 1);
        }

        dates.add(new RoomReportDate(enddate));

        return dates;
    }

}
