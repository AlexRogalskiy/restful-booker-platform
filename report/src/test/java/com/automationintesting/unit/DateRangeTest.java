package com.automationintesting.unit;

import api.DateRange;
import model.report.RoomReportDate;
import org.approvaltests.Approvals;
import org.junit.Test;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class DateRangeTest {

    @Test
    public void testDateRange() throws ParseException {
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd");
        Date checkin = simpleDateFormat.parse("2018-01-01");
        Date checkout = simpleDateFormat.parse("2018-01-05");

        List<RoomReportDate> dateList = new ArrayList<>();

        dateList = DateRange.parse(dateList, checkin, checkout);

        Approvals.verify(dateList);
    }

}
