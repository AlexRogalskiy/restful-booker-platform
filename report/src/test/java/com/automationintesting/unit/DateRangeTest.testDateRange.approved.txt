[RoomReportDate{date=2018-01-01}, RoomReportDate{date=2018-01-02}, RoomReportDate{date=2018-01-03}, RoomReportDate{date=2018-01-04}, RoomReportDate{date=2018-01-05}]